src/mmw/apps/bigcz/clients/cinergi/search.py

```diff
@@ -120,6 +121,21 @@ def prepare_time(from_date, to_date):
     return value
 
 
+def prepare_query(query):
+    """
+    Prepare query to always AND all search terms
+
+    Spaces will be replaced with AND. Existing ANDs and ORs will be preserved.
+    """
+    def intersperse_and(phrase):
+        return ' AND '.join([w for w in phrase.split()
+                             if not re.match('and', w, flags=re.IGNORECASE)])
+
+    factors = re.split(' or ', query, flags=re.IGNORECASE)
+
+    return ' OR '.join(map(intersperse_and, factors))
```

Comment:
Earlier I tried writing it as:

```python
return ' OR '.join([' AND '.join(w for w in f.split()
                                 if not re.match('and', w,
                                                 flags=re.IGNORECASE))
                    for f in factors])
```

but thought it less clear. Any suggestions for improving readability welcome.

Reply:
Maybe if I replace `factors` with `phrases` it'll be a little nicer.

Reply:
This is a small improvement, but if the API is not case sensitive, you could convert everything to lower/uppercase beforehand which would let you drop `flags=re.IGNORECASE`.

Reply:
Just checked, the API is _not_ case sensitive. Will replace the `re` stuff and that should help make it easier to read. Will add comments if it's still not clear enough. Thanks